Feat: Filter functionality on Product Page

abhijitdotsharma opened the pull request:
Following functionality added 
- Sort
- Category Select
- Price Slider
- Rating

Important Files to Check
- `src/pages/product-page`
- `src/components/filter`
- `src/components/products`
- `src/context/product-context.js`
- `src/reducer/filterReducer.js`
- `src/reducer/utilities`

Files not to check - 
- Mock Backend files were added in this branch, so they are included in this PR, can be skipped
- All the UI related code was reviewed in this PR  #2  .
But any review, would still be appreciated

abhijitdotsharma committed:
install: react-router-dom added

abhijitdotsharma committed:
add: Routes added to navigate home and product page

abhijitdotsharma committed:
add: BrowserRouter wrapped to App component

abhijitdotsharma committed:
refactor: a tags replaced with Link (react router)

abhijitdotsharma committed:
ProductPage imported with dummy text

abhijitdotsharma committed:
create: product-page.css

abhijitdotsharma committed:
create: ProductPage.jsx created with dummy text

abhijitdotsharma committed:
build: mockbee and uuid dependencies added

abhijitdotsharma committed:
remove: removed empty default files

abhijitdotsharma committed:
feat: added mock backend mockbee

abhijitdotsharma committed:
route for mockman added

abhijitdotsharma committed:
feat: product page with navigation filter and products component

abhijitdotsharma committed:
temporary link to mockman added

abhijitdotsharma committed:
temporary link to mockman added

abhijitdotsharma committed:
add: Products.jsx component

abhijitdotsharma committed:
add - Products .jsx css file

abhijitdotsharma committed:
update - components/index.js

abhijitdotsharma committed:
minor - renamed & to and in UI

abhijitdotsharma committed:
Create - filter.css

abhijitdotsharma committed:
update - db/products.js with picsum links

abhijitdotsharma committed:
removed temporary link to landing page

abhijitdotsharma committed:
Add - dummy products

abhijitdotsharma committed:
minor - updated a tag with Link tag

abhijitdotsharma committed:
create - product-context file

abhijitdotsharma committed:
refactored Products.jsx to use ProductCard component

abhijitdotsharma committed:
adjustment of margins, temporary solution

abhijitdotsharma committed:
ProductProvider  wrapped to App

abhijitdotsharma committed:
Create - ProductCard component to render individual products

abhijitdotsharma committed:
Merge pull request #2 from abhijitdotsharma/product-page-UI

Product page UI

abhijitdotsharma committed:
updated categories

abhijitdotsharma committed:
updated products

abhijitdotsharma committed:
Feat: Filter Functionality

abhijitdotsharma committed:
update- ProductCard to render individual products

abhijitdotsharma committed:
update- product.css

abhijitdotsharma committed:
udpate - Products.jsx to render ProductCard

abhijitdotsharma committed:
ProductPage with Reducer logic and different components

Reducer Logic will be seperated into different file in next PR along with CSS

abhijitdotsharma committed:
create - filterReducer

abhijitdotsharma committed:
Fix - card__info div taking more margin

used * selector in products.css card class to
give margin,padding = 0
box-sizing changed to border-box

abhijitdotsharma committed:
removed comments

abhijitdotsharma committed:
update Products.jsx

abhijitdotsharma committed:
update-ProductPage.jsx

abhijitdotsharma committed:
minor - removed comments and extra code

abhijitdotsharma committed:
Update product-context.js

abhijitdotsharma committed:
refactor - reducer logic moved to filterReducer.js

abhijitdotsharma committed:
minor - productReducer import renamed to filterReducer

abhijitdotsharma committed:
create - utilites for filterReducer

munjalSaksham71 commented on src/components/filter/Filter.jsx:
```diff
@@ -0,0 +1,134 @@
+            <div class="filter-container">
+                <div class="filter">
+                    <h3>Filter</h3>
+                    <span onClick={() => clearFilters()}>Clear</span>
```
```onClick={clearFilter}```

abhijitdotsharma replied:
Didn't realize this, thanks for pointing out

munjalSaksham71 commented on src/components/filter/Filter.jsx:
```diff
@@ -0,0 +1,134 @@
+                <div class="price">
+                    <h3>Price</h3>
+                    <div class="slider">
+                        {clearSliderBtn && <button onClick={() => filterDispatchAndHideClear()}>clear</button>}
```
same like above, If you are not passing any thing in arguments

munjalSaksham71 commented on src/components/filter/Filter.jsx:
```diff
@@ -0,0 +1,134 @@
+
+                <div class="category">
+                    <h3>Category</h3>
+                    <label for="">
```
in React, it is htmlFor="" and please try to give some value inside it.

munjalSaksham71 commented on src/components/filter/Filter.jsx:
```diff
@@ -0,0 +1,134 @@
+                        />
+                        4 stars and above
+                    </label>
+                    {console.log("productState.rating: ", productState.rating)}
```
Remove Log

munjalSaksham71 commented on src/components/products/Products.jsx:
```diff
@@ -0,0 +1,16 @@
+import "./products.css";
+import { ProductCard} from "./ProductCard";
+
+export default function Products(props) {
```
Try to be consistent. Either use traditional Functions or arrow function

abhijitdotsharma replied:
sure, I will be mindful next. Thanks

munjalSaksham71 commented on src/context/product-context.js:
```diff
@@ -0,0 +1,28 @@
+            const res = await axios.get("/api/products/");
+            const resProducts = res.data.products;
+
+            setProducts(prev => [...prev, ...resProducts]);
```
Why are u using prev here. I guess you don't need here.

munjalSaksham71 commented on src/components/filter/Filter.jsx:
```diff
@@ -0,0 +1,134 @@
+                        <label>
+                            <input
+                                type="range"
+                                min="0"
```
Give some defaultValue.

munjalSaksham71 reviewed (approved):
Suggest some changes - Please go through them. 
One more thing in UI you can be more consistent in gaps b/w the checkbox and text.

abhijitdotsharma committed:
cleanup

- removed for in label tag
- refactored no argument callback functions
- removed console.log
- gave default value to slider

abhijitdotsharma committed:
refactor - setProducts didn't need prev state, changed it

mariyasada commented on src/context/product-context.js:
```diff
@@ -0,0 +1,28 @@
+
+    useEffect(()=>{
+        (async () => {
+            const res = await axios.get("/api/products/");
+            const resProducts = res.data.products;
+
+            setProducts(resProducts);
```
you can destructure data here like {data} then setproducts(data.products)

mariyasada reviewed (approved):
liked that you have written a comment on a particular line of code what it actually does.good to go

abhijitdotsharma committed:
refactor - destructured axios call
